pylint/reporters/text.py

```diff
@@ -183,13 +184,27 @@ def __init__(self, output: Optional[TextIO] = None) -> None:
         super().__init__(output)
         self._modules: Set[str] = set()
         self._template = self.line_format
+        self._checked_template = False
 
     def on_set_current_module(self, module: str, filepath: Optional[str]) -> None:
         self._template = str(self.linter.config.msg_template or self._template)
 
     def write_message(self, msg: Message) -> None:
         """Convenience method to write a formatted message with class default template"""
-        self.writeln(msg.format(self._template))
+        self_dict = msg._asdict()
+        for key in ("end_line", "end_column"):
+            self_dict[key] = self_dict[key] or ""
+
+        # Check to see if all parameters in the template are attributes of the Message
+        if not self._checked_template:
+            template = self._template
+            parameters = re.findall(r"\{(.+?)\}", template)
+            for parameter in parameters:
+                if parameter not in self_dict:
+                    template = template.replace(f"{{{parameter}}}", "")
```

Comment:
Is it save to overwrite `self._template` directly?

Reply:
It should be in its new place